src/main/java/com/synerzip/ExpenseCalculation/Category.java

```diff
@@ -0,0 +1,54 @@
+package com.synerzip.ExpenseCalculation;
+
+import java.util.ArrayList;
+import java.util.List;
+
+import javax.persistence.Column;
+import javax.persistence.Entity;
+import javax.persistence.FetchType;
+import javax.persistence.GeneratedValue;
+import javax.persistence.GenerationType;
+import javax.persistence.Id;
+import javax.persistence.OneToMany;
+import javax.persistence.TableGenerator;
+
+@Entity
+@TableGenerator(name="tab", initialValue=0, allocationSize=50)
+public class Category {
+	@Id
+	@GeneratedValue(strategy = GenerationType.TABLE,generator="tab")
+	@Column(nullable=false)
+	private int id;
```

Comment:
why you are using GenerationType as TABLE instead of AUTO or IDENTITY?
why  'nullable=false' if declared as @Id?